### PR TITLE
remove compiler warnings, whitespace cleanup

### PR DESCRIPTION
**Pull request by svenevs**
- Introduces `NABO_UNUSED` macro to bypass -Wunused-parameter
  in the trap methods
- CXX11: `register` is deprecated, removed
- Strip trailing whitespace on all source files
    - I can undo this and just revive the stuff related to compiler warnings, I've got my editor setup to do this by default so it felt weird to submit a patch with just `nabo.h` and `index_heap.h` getting changed but not the others.  If you would like the compiler fixes without the whitespace stuff, let me know!

### CODE REVIEW

File: nabo/index_heap.h [hunk 1]
```diff
@@ -217,17 +217,17 @@ namespace Nabo
...
 		inline const VT& headValue() const { return data[0].value; }
-		
+
 		//! replace the largest value of the heap
 		/** \param index new point index
 		 * 	\param value new distance value */
 		inline void replaceHead(const Index index, const Value value)
 		{
-			register size_t i = 0;
+			size_t i = 0;
```
Comment (svenevs) [inline nabo/index_heap.h]: Removed `register` (before `size_t i = 0`) from docs.

File: nabo/index_heap.h [hunk 2]
```diff
@@ -311,17 +311,17 @@ namespace Nabo
...
 		inline const VT& headValue() const { return headValueRef; }
-		
+
 		//! replace the largest value of the heap
 		/** \param index new point index
 		 * 	\param value new distance value */
 		inline void replaceHead(const Index index, const Value value)
 		{
-			register size_t i;
+			size_t i;
```
Comment (svenevs) [inline nabo/index_heap.h]: Removed `register` before `size_t i`.

File: nabo/nabo.h [hunk 3]
```diff
@@ -203,12 +203,19 @@ libnabo differs from \ref ANN on the following points:
 
 */
 
+/**
+ * Used to avoid unused parameter compiler warnings, while still being able to
+ * document the parameters with Doxygen.  Example usage: `NABO_UNUSED(var);`
+ * -- the `;` is required after the macro invocation.
+ */
+#define NABO_UNUSED(var) (void) var
```
Comment (svenevs) [inline nabo/nabo.h]: Add `NABO_UNUSED` macro.

File: nabo/nabo.h [hunk 4]
```diff
@@ -386,13 +393,19 @@ namespace Nabo
...
 
 		//! Prevent creation of trees with the wrong matrix type. Currently only dynamic size matrices are supported.
 		template <typename WrongMatrixType>
 		static NearestNeighbourSearch* create(const WrongMatrixType& cloud, const Index dim = std::numeric_limits<Index>::max(), const SearchType preferedType = KDTREE_LINEAR_HEAP, const unsigned creationOptionFlags = 0, const Parameters& additionalParameters = Parameters())
 		{
+		  NABO_UNUSED(cloud);
+		  NABO_UNUSED(dim);
+		  NABO_UNUSED(preferedType);
+		  NABO_UNUSED(creationOptionFlags);
+		  NABO_UNUSED(additionalParameters);
```
Comment (svenevs) [inline nabo/nabo.h]: trap method 1 gets `NABO_UNUSED`

File: nabo/nabo.h [hunk 5]
```diff
@@ -402,6 +415,10 @@ namespace Nabo
 		template <typename WrongMatrixType>
 		static NearestNeighbourSearch* createBruteForce(const WrongMatrixType& cloud, const Index dim = std::numeric_limits<Index>::max(), const unsigned creationOptionFlags = 0)
 		{
+		  NABO_UNUSED(cloud);
```
Comment (svenevs) [inline nabo/nabo.h]: trap method 2 gets `NABO_UNUSED`

File: nabo/nabo.h [hunk 6]
```diff
@@ -411,6 +428,11 @@ namespace Nabo
 		template <typename WrongMatrixType>
 		static NearestNeighbourSearch* createKDTreeLinearHeap(const WrongMatrixType& cloud, const Index dim = std::numeric_limits<Index>::max(), const unsigned creationOptionFlags = 0, const Parameters& additionalParameters = Parameters())
 		{
+		  NABO_UNUSED(cloud);
```
Comment (svenevs) [inline nabo/nabo.h]: Trap method 3 gets `NABO_UNUSED`

File: nabo/nabo.h [hunk 7]
```diff
@@ -420,35 +442,39 @@ namespace Nabo
 		template <typename WrongMatrixType>
 		static NearestNeighbourSearch* createKDTreeTreeHeap(const WrongMatrixType&, const Index dim = std::numeric_limits<Index>::max(), const unsigned creationOptionFlags = 0, const Parameters& additionalParameters = Parameters())
 		{
+		  NABO_UNUSED(dim);
```
Comment (svenevs) [inline nabo/nabo.h]: trap method 4 gets `NABO_UNUSED`

**Comment by HannesSommer** [comment]
okay to test

**Comment by HannesSommer** [comment]
test this please

**Pull request closed without merging**